langchain/src/callbacks/base.ts

```diff
@@ -93,21 +131,43 @@ export abstract class BaseCallbackManager extends BaseCallbackHandler {
 export class CallbackManager extends BaseCallbackManager {
   handlers: BaseCallbackHandler[];
 
-  constructor() {
+  name = "callback_manager";
+
+  private _currentRunId?: string;
+
+  private readonly _parentRunId?: string;
+
+  constructor(currentRunId?: string, parentRunId?: string) {
     super();
     this.handlers = [];
+    this._currentRunId = currentRunId;
+    this._parentRunId = parentRunId;
+  }
+
+  get currentRunId(): string | undefined {
+    return this._currentRunId;
+  }
+
+  // needed to avoid ESLint no-param-reassign error
+  setCurrentRunId(runId: string | undefined) {
```

Comment:
IMO this method should not exist, and currentRunId should be immutable after set in the constructor, as we discussed. If currentRunId needs to change for any reason a new instance should be created

Reply:
Yeah that's fair. I'm not too worried about the currentRunId changing around a lot as each time, before the callbackManager is used, it's cloned. But what you suggested sounds like a better abstraction